src/string/utf16.cr

```diff
@@ -12,22 +12,30 @@ class String
   # "hi 𐂥".to_utf16 # => Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # ```
   def to_utf16 : Slice(UInt16)
-    size = 0
+    if ascii_only?
+      # size == bytesize, so each char fits in one UInt16
+      return to_slice.map &.to_u16
+    end
+
+    # size < bytesize, so we need to count the number of characters that are
+    # two UInt16 wide.
+    u16_size = 0
     each_char do |char|
-      size += char.ord < 0x10000 ? 1 : 2
+      u16_size += char.ord < 0x1_0000 ? 1 : 2
```

Comment:
While I like the usage of `_`, why did you choose 4 '0's for the location of it?

Reply:
That's the typical grouping for hexadecimal numbers.

In this specific case it's even particularly useful because we're talking about UInt16 values which happen to fit exactly 4 hex digits.

Reply:
Thanks for that...I've seen the `_` used a ton with decimal digits before. but I've never seen hex digits broken up, even though I did know it was possible.